contracts/starknet/safe_math.cairo

```diff
@@ -123,40 +123,9 @@ func _mul{range_check_ptr, bitwise_ptr: BitwiseBuiltin*}(lhs: Uint256, rhs: Int2
     return (res,);
 }
 
-// function _min(uint256 x, uint256 y) internal pure returns (uint256 z) {
-//         z = x <= y ? x : y;
-//     }
-func min{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
-    x: Uint256, y: Uint256
-) -> (z: Uint256) {
-    let (x_le: felt) = uint256_le(x, y);
-    if (x_le == 1) {
-        return (z=x);
-    } else {
-        return (z=y);
-    }
-}
-
-func _uint_to_felt{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
-    value: Uint256
-) -> (value: felt) {
-    assert_lt_felt(value.high, 2 ** 123);
-    return (value.high * (2 ** 128) + value.low,);
-}
-
-func _felt_to_uint{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
-    value: felt
-) -> (value: Uint256) {
-    let (high, low) = split_felt(value);
-    tempvar res: Uint256;
-    res.high = high;
-    res.low = low;
-    return (res,);
-}
-
-func div{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+func div_rem{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
     a: Uint256, b: Uint256
-) -> (c: Uint256) {
+) -> (c: Uint256, rem: Uint256) {
     alloc_locals;
     uint256_check(a);
```

Comment:
Do we need to check uint256 here? I would assume that uints should be checked for correctness "at the border" when they eneter the contract.

Reply:
hm not sure, we dont always check storage vars as we read them. Seems safer to check them before doing the operation or could lead to unexpected results.

Reply:
I would assume that uints from storage are well formed